Sources/Kafka/KafkaConsumer.swift

```diff
@@ -492,6 +492,27 @@ public final class KafkaConsumer: Sendable, Service {
     ///     - message: Last received message that shall be marked as read.
     /// - Throws: A ``KafkaError`` if committing failed.
     public func scheduleCommit(_ message: KafkaConsumerMessage) throws {
+        try scheduleCommit(topic: message.topic,
+                           partition: message.partition,
+                           offset: message.offset)
+    }
+
+    /// Mark all messages up to the passed message in the topic as read.
+    /// Schedules a commit and returns immediately.
+    /// Any errors encountered after scheduling the commit will be discarded.
+    ///
+    /// This method is only used for manual offset management.
+    ///
+    /// - Warning: This method fails if the ``KafkaConsumerConfiguration/isAutoCommitEnabled`` configuration property is set to `true` (default).
+    ///
+    /// - Parameters:
+    ///     - topic: Topic where the message that should be marked as read resides.
+    ///     - partition: Partition where the message that should be marked as read resides.
+    ///     - offset: Offset of the message that shall be marked as read.
+    /// - Throws: A ``KafkaError`` if committing failed.
+    public func scheduleCommit(topic: String,
+                               partition: KafkaPartition,
```

Comment:
Wondering if we have to expose both a topic and a partition argument here. Topic we should be able to get from our config and the consumer should always stick to the same topic. Partition is a bit more interesting due to rebalancing events.

NIT: Could you also fix the line breaks in this whole commit. We normally line break after the `(` and before `)`. This is matching what Xcode does.

Reply:
A client can be configured to consume from multiple topics and partitions (see `assign/1`) so I do not think there is a way around passing the topic in explicitly.

Reply:
Right now a client can only be configured to consume from a single topic since we are not publicly exposing `assign`. Following this reasoning we shouldn't expose `topic` here either but take it from the config,